sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/util/ModelHelper.java

```diff
@@ -208,6 +210,14 @@ public static ShareFileItemProperties transformFileProperty(FileProperty propert
             property.getLastWriteTime(), property.getChangeTime(), property.getLastModified(), property.getEtag());
     }
 
+    public static List<ShareFileHandleAccessRights> transformAccessRightsList(List<AccessRight> accessRightList) {
+        List<ShareFileHandleAccessRights> result = new ArrayList<>();
+        accessRightList.forEach(item -> {
+            result.add(ShareFileHandleAccessRights.fromString(item.toString()));
+        });
+        return result;
```

Comment:
Since both types are enums let's do a manual mapping so we don't have to do an enum -> string -> enum conversion. If the off case this becomes a hot path this is much slower.

Longer term we can look at having code generation handling the rename so we don't need a conversion layer at all.